model/fingerprinting_test.go

```diff
@@ -0,0 +1,214 @@
+// Copyright 2013 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"sort"
+	"testing"
+)
+
+func TestFingerprintFromString(t *testing.T) {
+	fs := "4294967295"
+
+	f, err := FingerprintFromString(fs)
+
+	if err != nil {
+		t.Errorf("unexpected error while getting Fingerprint from string: %s", err.Error())
+	}
+
+	expected := Fingerprint(285960729237)
+
+	if expected != f {
+		t.Errorf("expected to get %d, but got %d instead", f, expected)
+	}
+
+	f, err = ParseFingerprint(fs)
+
+	if err != nil {
+		t.Errorf("unexpected error while getting Fingerprint from string: %s", err.Error())
+	}
+
+	if expected != f {
+		t.Errorf("expected to get %d, but got %d instead", f, expected)
+	}
+}
+
+func TestFingerprintsSort(t *testing.T) {
+	fingerPrints := Fingerprints{
+		14695981039346656037,
+		285960729237,
+		0,
+		4294967295,
+		285960729237,
+		18446744073709551615,
+	}
+
+	sort.Sort(fingerPrints)
+
+	expected := Fingerprints{
+		0,
+		4294967295,
+		285960729237,
+		285960729237,
+		14695981039346656037,
+		18446744073709551615,
+	}
+
+	for i, f := range fingerPrints {
+		if f != expected[i] {
+			t.Errorf("expected Fingerprint %d, but got %d for index %d", expected[i], f, i)
+		}
+	}
+}
+
+func TestFingerprintSet(t *testing.T) {
+	//sets of unequal length
```

Comment:
Comments should start with a capital letter and end with a full stop

Reply:
Done